sequencer/src/persistence.rs

```diff
@@ -1316,83 +1324,138 @@ mod persistence_tests {
+            async move {
+                {
+                    setup_stake_table_contract_for_test(
+                        l1_url,
+                        &deployer,
+                        st_addr,
+                        token_addr,
+                        validators,
+                        DelegationConfig::MultipleDelegators,
+                    )
+                    .await
+                    .expect("stake table setup failed");
+                }
+            }
+        });
 
-        //start the network
-        let _test_network = TestNetwork::new(testnet_config, PosVersion::new()).await;
-        let client: Client<ServerError, SequencerApiVersion> = Client::new(
-            format!("http://localhost:{query_service_port}")
-                .parse()
-                .unwrap(),
-        );
-        client.connect(None).await;
-        tracing::info!(query_service_port, "server running");
+        let storage = P::tmp_storage().await;
+        let persistence = P::options(&storage).create().await.unwrap();
 
-        // wait until we enter in epoch 3
-        let _initial_blocks = client
-            .socket("availability/stream/blocks/0")
-            .subscribe::<BlockQueryData<SeqTypes>>()
-            .await
-            .unwrap()
-            .take(60)
-            .try_collect::<Vec<_>>()
-            .await
-            .unwrap();
+        let l1_client = L1ClientOptions {
+            stake_table_update_interval: Duration::from_secs(7),
+            l1_retry_delay: Duration::from_millis(10),
+            l1_events_max_block_range: 10000,
+            ..Default::default()
+        }
+        .connect(vec![l1_url])
+        .unwrap();
+        l1_client.spawn_tasks().await;
 
-        // Load initial persisted events and validate they exist.
-        let persistence = persistence_options[0].clone().create().await.unwrap();
-        let res1 = persistence
-            .load_events()
-            .await
-            .expect("failed to load events");
-        assert!(res1.is_some());
+        let fetcher = StakeTableFetcher::new(
+            Arc::new(NullStateCatchup::default()),
+            Arc::new(Mutex::new(persistence.clone())),
+            l1_client.clone(),
+            ChainConfig {
+                stake_table_contract: Some(st_addr),
+                base_fee: 0.into(),
+                ..Default::default()
+            },
+        );
 
-        let (mut prev_l1, prev_events) = res1.unwrap();
-        assert!(!prev_events.is_empty());
+        // sleep so that we have enough events
+        sleep(Duration::from_secs(20)).await;
 
+        fetcher.spawn_update_loop().await;
+        let mut prev_l1_block = 0;
         for _i in 0..10 {
             // Wait for more than update interval to assert that persistence was updated
-            // L1 update interval is 5s for testing
-            tokio::time::sleep(std::time::Duration::from_secs(6)).await;
+            // L1 update interval is 2s in this test
```

Comment:
Where is that `2s` set? The stake table update interval is 7s and the L1 block time is 1s.